src/App.tsx

```diff
@@ -15,47 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  none,
+  alphabet,
+  length,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((item1, item2) => {
+    switch (sortType) {
+      case SortType.alphabet:
+        return item1.localeCompare(item2);
+      case SortType.length:
+        return item1.length - item2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.none);
+
+  const goods = getReorderedGoods(goodsFromServer, { isReversed, sortType });
+
+  const sortByAlphabet = () => {
+    setSortType(SortType.alphabet);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.length);
+  };
+
+  const reverse = () => {
+    setReversed(!isReversed);
+  };
+
+  const reset = () => {
+    setSortType(SortType.none);
+    setReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button is-info',
+            { 'is-light': sortType !== SortType.alphabet },
+          )}
+          onClick={sortByAlphabet}
         >
           Sort alphabetically
         </button>
```

Comment:
U use this logic several times - let's move it to a separate function

Reply:
you can create a component for button and use this component for buttons, like Anna did in the lecture with PeopleTable